### PR TITLE
Silence checkpointing

### PR DESCRIPTION
**Pull request by mlange05**
Adds a decorator to temporarily change log levels and uses it to silence checkpointing output. Also updates `flake8` rules.

### CODE REVIEW

**Comment by navjotk** [comment]
Shouldn't be required if raising the log-level to WARNING, no?

File: devito/tools.py [hunk 1]
```diff
@@ -307,3 +307,22 @@ def infer_cpu():
     if platform not in configuration._accepted['platform']:
         platform = default_platform
     return isa, platform
+
+
+class silencio(object):
```
Comment (FabioLuporini) [inline devito/tools.py]: in ``tools.py`` or ``logger.py`` ?

File: .travis.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ env:
   global:
   - ENCRYPTION_LABEL: "1909213e8aea"
   - COMMIT_AUTHOR_EMAIL: "pandolfovince@gmail.com"
-  - DEVITO_LOGGING: "DEBUG"
+  - DEVITO_LOGGING: "INFO"
```
Comment (navjotk) [inline .travis.yml]: INFO or WARNING?

**Comment by FabioLuporini** [comment]
In fact, thinking again about it, why not just stick to WARNING everywhere ? the ``silencio`` util may still be useful though for application code.

**Comment by mloubout** [comment]
>the silencio util may still be useful though for application code.

isn't what "ERROR" does?

**Comment by FabioLuporini** [comment]
> isn't what "ERROR" does?

I'm not sure I understand what you mean by that. If you're using any of the Devito high level wrappers, or you simply want to silence Devito in a particular region, then you can use this utility

File: .travis.yml (hunk 2 above)
Comment (mlange05) [reply .travis.yml]: I strongly believe it should be `INFO`, since that mirrors the default user experience.

**Comment by FabioLuporini** [comment]
> I strongly believe it should be INFO, since that mirrors the default user experience.

OK, I really don't have strong thoughts in merit, so I'm happy to switch to INFO. I think the major problems in the end were:
- useless DSE/DLE messages (gone time ago)
- excessive pollution from various tests (fixed through ``silencio``)

**Comment by navjotk** [comment]
The next biggest source of log clutter after this are the compiler warnings.

**Comment by FabioLuporini** [comment]
All good, merging...

**Comment by FabioLuporini** [comment]
Merged.

**Pull request merged**